feat(dashmate): high-performance nodes registration with `setup` command (#794)
dashpay/platform: packages/dashmate/README.md

The CLI can be used to perform routine tasks. Invoke the CLI with `dashmate` if linked during installation, or with `node bin/dashmate` if not linked. To list available commands, either run `dashmate` with no parameters or execute `dashmate help`. To list the help on any command just execute the command, followed by the `--help` option.

### Set up node

The `setup` command is used to quickly configure common node configurations. Arguments may be provided as options, otherwise they will be queried interactively with sensible values suggested.

```
Set up a new Dash node

USAGE
  $ dashmate setup [PRESET] [-v] [-d] [-c <value>] [-m <value>]

ARGUMENTS
  PRESET  (mainnet|testnet|local) Node configuration preset

FLAGS
  -c, --node-count=<value>      number of nodes to set up
  -d, --[no-]debug-logs         enable debug logs
  -m, --miner-interval=<value>  interval between blocks
  -v, --verbose                 use verbose mode for output

DESCRIPTION
  Set up a new Dash node
```

Supported presets:

>  * `testnet` - a masternode or full node for testnet

Reviewer: Should be any entry here for `mainnet` now too?

Author: Yeah, please update readme )))

### Node groups

> CLI allows to [setup](#set-up-node) and operate multiple nodes. Only the `local` preset is supported at the moment.

Reviewer: ```suggestion
CLI allows [setup](#set-up-node) and operation of multiple nodes. Only the `local` preset is supported at the moment.
```

#### Default group

> The [setup](#set-up-node) command set corresponding group as default. To output the current default group or set another one as default use `group:default` command.

Reviewer: ```suggestion
The [setup](#set-up-node) command sets the corresponding group as default. To output the current default group or set another one as default use the `group:default` command.
```

```
USAGE
  $ dashmate group stop [-v] [--group <value>] [-f]

FLAGS

>   -f, --force      force stop even if any is running

Reviewer 2: Can you change this one to service?

dashmate group default testnet
```

Ports and other required options need to be updated to avoid port collisions before starting the group of nodes.

### Development

> To start a local dash network, the `setup` command with the `local` preset can be used to generate configs, mine some tDash, register masternodes and populate the nodes with the data required for local development.

Reviewer: @strophy It looks like we have a serial comma in the re-index section so one might be needed here.
```suggestion
To start a local dash network, the `setup` command with the `local` preset can be used to generate configs, mine some tDash, register masternodes, and populate the nodes with the data required for local development.
```

`dashmate group:stop --force` to stop group of nodes (local)

#### Running services detected. Please ensure all services are stopped for this config before starting
Some nodes are still running and preventing dashmate from starting properly. This may occur after a command exits with an error. Try to stop force stop the nodes using the `--force` option before trying to run the `start` command again.

`dashmate stop --force` to stop single node (fullnode / masternode)

`dashmate group:stop --force` to stop group of nodes (local)

#### externalIp option is not set in base config

> This may happen when you switch back and forth between major versions, making config incompatible. In this case, do a manual reset and run setup again

Reviewer: ```suggestion
This may happen when you switch back and forth between major versions, making config incompatible. In this case, do a manual reset and run setup again.
```
